pylint/checkers/typecheck.py

```diff
@@ -172,6 +172,7 @@ def _string_distance(seq1: str, seq2: str, seq1_length: int, seq2_length: int) -
     return row[seq2_length - 1]
 
 
+@lru_cache(maxsize=256)
```

Comment:
Can we add this to clear_lru_caches()?

Reply:
This created a circular import between typecheck and utils so I created a new file.